src/Toml/Bi/Combinators.hs

```diff
@@ -250,9 +213,13 @@ maybeP converter key = let bi = converter key in Bijection
     }
   where
     handleNotFound :: DecodeException -> Env (Maybe a)
-    handleNotFound (KeyNotFound _)   = pure Nothing
-    handleNotFound (TableNotFound _) = pure Nothing
-    handleNotFound e                 = throwError e
+    handleNotFound e@(KeyNotFound name)
+        | name == key = pure Nothing
+        | otherwise = throwError e
+    handleNotFound e@(TableNotFound name)
+        | name == key = pure Nothing
+        | otherwise = throwError e
```

Comment:
Probably we could some how get rid of code duplication.. But I can't see the better way for this, so let keep it as it is. At least it's readable!

Reply:
I think it's possible. Will try to rewrite it.